src/components/NewActivityList.js

```diff
@@ -0,0 +1,147 @@
+import React, { Component } from 'react'
+import { Table, Button, Modal, ControlLabel, FormControl } from 'react-bootstrap'
+import './components.css'
+import firebase from 'firebase'
+import toastr from 'toastr'
+import moment from 'moment'
+import CountDown from './CountDown'
+
+class NewActivityList extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      hours: '',
+      money: '',
+      how_long: true,
+      startedAt: '',
+      isModalShown: true
+    }
+  }
+
+  componentDidMount() {
+    const userId = firebase.auth().currentUser.uid;
+    firebase.database().ref('/users/' + userId + '/settings').once('value').then(snapshot => {
+      const currentRate = (snapshot.val().currentRate * 0.4)
+      this.setState({
+        money: currentRate
+      })
+    })
+    firebase.database().ref('/users/' + userId + '/challenge').once('value').then(snapshot => {
+      if (snapshot.val().startedAt + 86400000 > moment()) {
+        this.setState({
+          hours: snapshot.val().hours,
+          money: snapshot.val().money,
+          how_long: snapshot.val().how_long,
+          startedAt: snapshot.val().startedAt,
+          isModalShown: false
+        })    
+      }
+    })
+  }
+
+  numberChecker(field, value) {
+    if (!isNaN(value)) {
+      this.setState({
+        [field]: value,
+      })
+    }
+  }
+
+  addChallenge = () => {
+    const { hours, money, how_long } = this.state
+    if ((hours.length > 0) && (money > 0)) {
+      this.setState({error: ''});
+      this.challengeSave({hours, money, how_long});
+    } else {
+      toastr.error('Please enter correct details');
+    }
+  }
+
+  challengeSave(params) {
+    const { hours, money, how_long } = params
+    const userUid = this.props.user.uid
+    const key = firebase.database().ref('days').push().key
+    firebase.database().ref('days/' + key).update({
+      hours,
+      money,
+      how_long,
+      userUid,
+      startedAt: Date.now(),
+      isDayOver: false,
+      isCharged: false
+    }).then(() => toastr.success('Your Activity saved!'))
```

Comment:
I'd prefer to have
.then() on new line
but it's not a mistake, just a bit more readable in my opinion